### ISSUE TITLE
339 rubocop offenses what am I doing wrong ?

### ISSUE THREAD

**Issue by username-is-already-taken2** [inspec/inspec#1876]
### Description

Hi @adamleff I don't know if you remember me from the chefconf hackday, on the day I was the one with windows laptop.

I shared with you that I wasn't able to get a clean `rake lint`, I know it ran fine on your mac and we chalked it up to a windows thing :)

The trouble is I'm not able to get a clean pass on either redhat or ubuntu, so clearly I'm doing something wrong :)

So I wondered if you would be willing to help me, so I can help you?  I'm trying to write an update for one of the resources and I would like to be able to run tests on my machine it possible.  Also if it is a setup thing then happy to write a blog post or add a documentation page to help pay it forward?

So these are the exact steps I took with the bento/ubuntu box, maybe you can see something I'm missing?

```
vagrant init bento/ubuntu-16.04; vagrant up --provider vmware_desktop
vagrant ssh

vagrant@vagrant:~$ curl -L https://omnitruck.chef.io/install.sh | sudo bash -s -- -P chefdk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 20500  100 20500    0     0  40522      0 --:--:-- --:--:-- --:--:-- 40513
ubuntu 16.04 x86_64
    <snip>   <snip>   <snip>
Setting up chefdk (1.4.3-1) ...
Thank you for installing Chef Development Kit!

vagrant@vagrant:~$ echo 'eval "$(chef shell-init bash)"' >> ~/.bashrc
vagrant@vagrant:~$ exit

vagrant ssh

vagrant@vagrant:~$ pwd
/home/vagrant

vagrant@vagrant:~$ chef verify
Running verification for component 'berkshelf'
    <snip>   <snip>   <snip>
Verification of component 'package installation' succeeded.

vagrant@vagrant:~$ git clone https://github.com/chef/inspec.git chef/inspec

vagrant@vagrant:~$ cd chef/inspec/

vagrant@vagrant:~/chef/inspec$ sudo gem update --system
Updating rubygems-update
Fetching: rubygems-update-2.6.12.gem (100%)
Successfully installed rubygems-update-2.6.12
    <snip>   <snip>   <snip>
RubyGems system software updated

vagrant@vagrant:~/chef/inspec$ gem update bundler

vagrant@vagrant:~/chef/inspec$ bundle install
Fetching https://github.com/chef/github-changelog-generator
Fetching gem metadata from https://rubygems.org/.......
Fetching version metadata from https://rubygems.org/..
Fetching dependency metadata from https://rubygems.org/.
Resolving dependencies....
Using rake 10.5.0
    <snip>   <snip>   <snip>
Bundle complete! 27 Gemfile dependencies, 126 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
Post-install message from minitar:
The `minitar` executable is no longer bundled with `minitar`. If you are
expecting this executable, make sure you also install `minitar-cli`.
Post-install message from httparty:
When you HTTParty, you must party hard!

vagrant@vagrant:~/chef/inspec$ rake lint
github_changelog_generator is not available.
Running RuboCop...
Inspecting 204 files
CCCCCCCCCCCCCCCWCCCCWCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC.C.C...C.CCCC/home/vagrant/chef/inspec/lib/inspec/dsl_shared.rb: Lint/Eval has the wrong namespace - should be Security
CCCCCCCCCCCCCCCCCCCCCCCCCCCCCC

Offenses:

Berksfile:1:1: C: Unnecessary utf-8 encoding comment.
# encoding: utf-8
^^^^^^^^^^^^^^^^^
    <snip>   <snip>   <snip>

www/tutorial/tutorial_files/inspec-mock/bin/inspec:2:1: C: Unnecessary utf-8 encoding comment.
# encoding: utf-8
^^^^^^^^^^^^^^^^^

204 files inspected, 339 offenses detected
RuboCop failed!
```

Any ideas? only thing I can think of is that it might be my shell?

**Comment by adamleff** [comment]
@username-is-already-taken2 of course I remember you :)

I'm honestly not sure what's going on, but I have to imagine it's your ruby environment.

Here are the steps I did... similar to yours but I skipped a few:

 * spun up a bento/ubuntu16.04 VM
 * Installed ChefDK
 * `echo 'eval "$(chef shell-init bash)"' >> ~/.bashrc` and logged back in
 * `git clone https://github.com/chef/inspec.git chef/inspec`
 * `cd chef/inspec`
 * `sudo yum install gcc make` since some native extensions will need to be built
 * `bundle install`
 * `bundle exec rake lint`

... and with those steps:

```
[vagrant@localhost inspec]$ bundle exec rake lint
/home/vagrant/chef/inspec/tasks/docs.rb:23: warning: already initialized constant PROJECT_DIR
/home/vagrant/chef/inspec/tasks/spdx.rb:18: warning: previous definition of PROJECT_DIR was here
Running RuboCop...
Inspecting 206 files
..............................................................................................................................................................................................................

206 files inspected, no offenses detected
```

**Comment by username-is-already-taken2** [comment]
Thanks for coming back to me,  i noticed you are using 'bundle exec rake lint' I was using just 'rake init'

I'll give that ago..... <newbie alert> need to really learn what 'bundle' is doing, don't really know when I need to use it :) </>

Get Outlook for iOS<https://aka.ms/o0ukef>
_____________________________

**Comment by adamleff** [comment]
Think of `bundler` as a virtual Ruby environment. Whenever you call `bundle exec` and pass a command, it will execute it within an isolated environment using only the gems (and the versions) specified in the Gemfile and `inspec.gemspec` file.

Without `bundle exec`, it will run with whatever is in your gem path, etc. which isn't necessarily what the project maintainers intended.

TL;DR: in daily use, you shouldn't ever have to worry about Bundler. However, when developing Ruby applications, if there's a Gemfile, you should use Bundler to make sure you're running with the maintainers' intended environment.

**Comment by username-is-already-taken2** [comment]
**Success !!** even on windows :) Thanks very much for your help

I might create a PR tomorrow to update the `readme.md` `Testing InSpec` section and add a bit about rake lint

**Comment by adamleff** [comment]
I'd love to see that PR! Glad you're squared away, too 🙂
